packages/modals/src/modal/index.tsx

```diff
@@ -1,37 +1,61 @@
-import { useRef, forwardRef, useCallback } from 'react';
-import PropTypes from 'prop-types';
+import {
+  forwardRef,
+  useCallback,
+  useRef,
+  MouseEvent,
+  TouchEvent,
+  PropsWithChildren,
+  HTMLAttributes,
+} from 'react';
 import clsx from 'clsx';
 import CrossIcon from '@goodhood/icons/lib/small/cross';
 
 import {
-  useScrollLock,
-  useTrack,
+  LegacyHandlers,
   useKeyManager,
   useLegacyHandle,
   useMisclickHandlers,
+  useScrollLock,
+  useTrack,
   useUnmount,
 } from './hooks';
 import { useModalProvider } from '../provider/hooks';
 import Portal from '../portal';
 
 import styles from './index.module.scss';
 
-const Modal = forwardRef(({
+export type ModalProps = PropsWithChildren<HTMLAttributes<HTMLDivElement> & {
```

Comment:
`HTMLDivElement` locks us in to `div` as root element. Changing the root node type would cause a breaking change in types